From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ambrus Toth <palpista11@gmail.com>
Date: Mon, 5 Oct 2026 17:53:29 +0200
Subject: [PATCH] Find tools in the bin directory of toolchains

In nixpkgs, the toolchain (the `swift` package) has its tools in `bin`
instead of `usr/bin`.
---
 Sources/SWBCore/ToolchainRegistry.swift | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Sources/SWBCore/ToolchainRegistry.swift b/Sources/SWBCore/ToolchainRegistry.swift
index 2421228..d6651dc 100644
--- a/Sources/SWBCore/ToolchainRegistry.swift
+++ b/Sources/SWBCore/ToolchainRegistry.swift
@@ -291,6 +291,8 @@ public final class Toolchain: Hashable, Sendable {
 
         var executableSearchPaths = [
             path.join("usr").join("bin"),
+            // nixpkgs: the toolchain (the `swift` package) has its tools in `bin` instead of `usr/bin`.
+            path.join("bin"),
         ]
 
         for platformExtension in pluginManager.extensions(of: PlatformInfoExtensionPoint.self) {
